--- SPQR/Makefile.orig	2011-12-07 23:20:52.000000000 +0100
+++ SPQR/Makefile	2012-01-07 15:26:12.640442839 +0100
@@ -4,7 +4,7 @@
 
 VERSION = 1.2.3
 
-include ../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 default: all
 
--- SPQR/Lib/Makefile.orig	2011-12-14 18:48:21.000000000 +0100
+++ SPQR/Lib/Makefile	2012-01-07 15:36:22.723796599 +0100
@@ -6,7 +6,7 @@
 
 ccode: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -14,13 +14,10 @@
 # check if METIS is requested and available
 ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
     # METIS is requested.  See if it is available
-    ifeq ($(METIS), $(wildcard $(METIS)))
-        ifeq (../../CAMD, $(wildcard ../../CAMD))
-            ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
+    ifeq ($(WITH_METIS), 1)
+	# assume CAMD and CCOLAMD are available
                 # METIS, CAMD, and CCOLAMD is available
                 CONFIG =
-            endif
-        endif
     endif
 endif
 #-------------------------------------------------------------------------------
@@ -91,7 +88,7 @@
 
 $(OBJ): $(INC)
 
-I = -I../../CHOLMOD/Include -I../../SuiteSparse_config -I../Include
+I = -I/usr/include/cholmod -I../Include
 
 C = $(CXX) $(CF) $(SPQR_CONFIG) $(CONFIG) $(I)
 
--- SPQR/Demo/Makefile.orig	2012-05-28 15:15:40.000000000 +0200
+++ SPQR/Demo/Makefile	2012-06-27 19:00:41.468480043 +0200
@@ -6,7 +6,7 @@
 
 ccode: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -15,24 +15,17 @@
 # check if METIS is requested and available
 ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
     # METIS is requested.  See if it is available
-    ifeq ($(METIS), $(wildcard $(METIS)))
-        ifeq (../../CAMD, $(wildcard ../../CAMD))
-            ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
-                # METIS, CAMD, and CCOLAMD is available
+    ifeq ($(WITH_METIS), 1)
+                # assume that METIS, CAMD, and CCOLAMD are available
                 CONFIG =
-                LIB_WITH_PARTITION = $(METIS) \
-                    ../../CCOLAMD/Lib/libccolamd.a \
-                    ../../CAMD/Lib/libcamd.a
-            endif
-        endif
+                LIB_WITH_PARTITION = $(METIS) -lccolamd -lcamd
     endif
 endif
 #-------------------------------------------------------------------------------
 
 CLIB = ../Lib/libspqr.a \
-        ../../SuiteSparse_config/libsuitesparseconfig.a \
-	../../CHOLMOD/Lib/libcholmod.a ../../AMD/Lib/libamd.a \
-	../../COLAMD/Lib/libcolamd.a $(LIB_WITH_PARTITION)
+	-lcholmod -lamd \
+	-lcolamd $(LIB_WITH_PARTITION)
 
 # use the BLAS and LAPACK defined by SuiteSparse_config.mk; do not use valgrind 
 FLIB = $(LAPACK) $(BLAS)
@@ -140,7 +133,7 @@
 	../Include/SuiteSparseQR_definitions.h \
 	../Include/SuiteSparseQR.hpp Makefile
 
-I = -I../../CHOLMOD/Include -I../../SuiteSparse_config -I../Include
+I = -I/usr/include/cholmod -I../Include
 
 C = $(CXX) $(CF) $(SPQR_CONFIG) $(CONFIG) $(I)
 
