--- SPQR/Makefile.orig	2011-12-07 23:20:52.000000000 +0100
+++ SPQR/Makefile	2012-01-07 15:26:12.640442839 +0100
@@ -4,7 +4,7 @@
 
 VERSION = 1.2.3
 
-include ../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 default: all
 
--- SPQR/Lib/Makefile.orig	2012-06-21 07:37:08.000000000 +0200
+++ SPQR/Lib/Makefile	2012-08-28 19:36:19.380393578 +0200
@@ -6,7 +6,7 @@
 
 ccode: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -15,18 +15,13 @@
 CONFIG = -DNPARTITION
 # check if METIS is requested and available
 ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
-    # METIS is requested.  See if it is available
-    ifeq ($(METIS_PATH), $(wildcard $(METIS_PATH)))
-        ifeq (../../CAMD, $(wildcard ../../CAMD))
-            ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
+    # METIS is requested.
+    ifeq ($(WITH_METIS), 1)
                 # METIS, CAMD, and CCOLAMD are available
-                LIB_WITH_PARTITION = $(METIS) ../../CCOLAMD/Lib/libccolamd.a \
-                    ../../CAMD/Lib/libcamd.a
-                I_WITH_PARTITION = -I$(METIS_PATH)/Lib \
-                    -I../../CCOLAMD/Include -I../../CAMD/Include
+                LIB_WITH_PARTITION = -lmetis -lccolamd -lcamd
+                I_WITH_PARTITION = -I/usr/include/metis \
+                    -I/usr/include/ccolamd -I/usr/include/camd
                 CONFIG =
-            endif
-        endif
     endif
 endif
 #-------------------------------------------------------------------------------
@@ -97,7 +92,7 @@
 
 $(OBJ): $(INC)
 
-I = -I../../CHOLMOD/Include -I../../SuiteSparse_config -I../Include
+I = -I/usr/include/cholmod -I../Include
 
 C = $(CXX) $(CF) $(SPQR_CONFIG) $(CONFIG) $(I)
 
--- SPQR/Demo/Makefile.orig	2012-06-21 07:47:51.000000000 +0200
+++ SPQR/Demo/Makefile	2012-08-28 19:38:04.650391386 +0200
@@ -6,7 +6,7 @@
 
 ccode: all
 
-include ../../SuiteSparse_config/SuiteSparse_config.mk
+include /usr/share/misc/SuiteSparse_config.mk
 
 #-------------------------------------------------------------------------------
 # the optional Partition module requires METIS, CAMD, and CCOLAMD
@@ -15,26 +15,20 @@
 CONFIG = -DNPARTITION
 # check if METIS is requested and available
 ifeq (,$(findstring -DNPARTITION, $(CHOLMOD_CONFIG)))
-    # METIS is requested.  See if it is available
-    ifeq ($(METIS_PATH), $(wildcard $(METIS_PATH)))
-        ifeq (../../CAMD, $(wildcard ../../CAMD))
-            ifeq (../../CCOLAMD, $(wildcard ../../CCOLAMD))
-                # METIS, CAMD, and CCOLAMD are available
-                LIB_WITH_PARTITION = $(METIS) ../../CCOLAMD/Lib/libccolamd.a \
-                    ../../CAMD/Lib/libcamd.a
-                I_WITH_PARTITION = -I$(METIS_PATH)/Lib \
-                    -I../../CCOLAMD/Include -I../../CAMD/Include
+    # METIS is requested.
+    ifeq ($(WITH_METIS), 1)
+                LIB_WITH_PARTITION = -lmetis -lccolamd -lcamd
+                I_WITH_PARTITION = -I/usr/include/metis \
+                    -I/usr/include/ccolamd -I/usr/include/camd
                 CONFIG =
-            endif
-        endif
     endif
 endif
 #-------------------------------------------------------------------------------
 
 CLIB = ../Lib/libspqr.a \
-        ../../SuiteSparse_config/libsuitesparseconfig.a \
-	../../CHOLMOD/Lib/libcholmod.a ../../AMD/Lib/libamd.a \
-	../../COLAMD/Lib/libcolamd.a $(LIB_WITH_PARTITION)
+        -lsuitesparseconfig \
+	-lcholmod -lamd \
+	-lcolamd $(LIB_WITH_PARTITION)
 
 # use the BLAS and LAPACK defined by SuiteSparse_config.mk; do not use valgrind 
 FLIB = $(LAPACK) $(BLAS)
@@ -145,7 +139,7 @@
 	../Include/SuiteSparseQR_definitions.h \
 	../Include/SuiteSparseQR.hpp Makefile
 
-I = -I../../CHOLMOD/Include -I../../SuiteSparse_config -I../Include
+I = -I/usr/include/cholmod -I../Include
 
 C = $(CXX) $(CF) $(SPQR_CONFIG) $(CONFIG) $(I)
 
